todo/views.py

```diff
@@ -45,11 +44,9 @@ def destroy(self, request, pk):
         return Response(status=status.HTTP_304_NOT_MODIFIED)
 
     def update(self, request, pk):
-        current_todo = Todo.objects.get(pk=pk)
-
-        if (request.user == current_todo.user):
-            changed_todo = TodoSerializer(current_todo, data=request.data)
-            if changed_todo.is_valid():
-                changed_todo.save()
-                return Response(changed_todo.data, status=status.HTTP_200_OK)
+        serializer = TodoSerializer(data=request.data)
+        if serializer.is_valid():
```

Comment:
serializer.is_valid(raise_exception=True) will automatically handle raise an exception so you don't have to do it manually down below (for example, if serializer.is_valid() is not True)